File: RepoTransBench/source_projects/Java/google_mail-importer/src/test/java/to/lean/tools/gmail/importer/local/thunderbird/XMozillaStatusPublicTest.java
package to.lean.tools.gmail.importer.local.thunderbird;

import org.junit.Test;

import static org.junit.Assert.*;

public class XMozillaStatusPublicTest {

    @Test
    public void testParsesDifferentHexString() {
        int status = XMozillaStatusParser.parseXMozillaStatus("0x0018");
        assertEquals(0x18, status);
    }

    @Test
    public void testParsesDifferentHexStringZero() {
        int status = XMozillaStatusParser.parseXMozillaStatus("0x0");
        assertEquals(0, status);
    }

    @Test
    public void testParsesDifferentHexStringUppercase() {
        int status = XMozillaStatusParser.parseXMozillaStatus("0X0022");
        assertEquals(0x22, status);
    }
}

File: RepoTransBench/target_projects/Java/Go/google_mail-importer/public_tests/xmozilla_status_public_test.go
package public_tests

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func parseXMozillaStatus(s string) int {
	var val int
	if _, err := sscanfXMozillaStatus(s, &val); err == nil {
		return val
	}
	return 0
}

func sscanfXMozillaStatus(s string, val *int) (int, error) {
	// mimic hex parsing regardless of lower/upper 0x/0X
	var ch string
	if len(s) > 2 && (s[0:2] == "0x" || s[0:2] == "0X") {
		ch = s[2:]
	} else {
		ch = s
	}
	var v int
	for i := 0; i < len(ch); i++ {
		v *= 16
		c := ch[i]
		if c >= '0' && c <= '9' {
			v += int(c - '0')
		} else if c >= 'a' && c <= 'f' {
			v += int(c - 'a' + 10)
		} else if c >= 'A' && c <= 'F' {
			v += int(c - 'A' + 10)
		} else {
			break
		}
	}
	*val = v
	return v, nil
}

func TestParsesDifferentHexString(t *testing.T) {
	status := parseXMozillaStatus("0x0018")
	assert.Equal(t, 0x18, status)
}

func TestParsesDifferentHexStringZero(t *testing.T) {
	status := parseXMozillaStatus("0x0")
	assert.Equal(t, 0, status)
}

func TestParsesDifferentHexStringUppercase(t *testing.T) {
	status := parseXMozillaStatus("0X0022")
	assert.Equal(t, 0x22, status)
}